izumi-reflect/izumi-reflect/src/test/scala/izumi/reflect/test/SharedTagTest.scala

```diff
@@ -1093,6 +1093,22 @@ abstract class SharedTagTest extends AnyWordSpec with XY[String] with TagAsserti
       assertCompiles("Tag[Array[Byte]]")
     }
 
+    "resolve parameters inside type bounds & wildcards" in { // Issue #31
+      def getUpperBoundTag[T: Tag] = Tag[Set[_ <: T]]
+
+      def getLowerBoundTag[T: Tag] = Tag[Set[_ >: T]]
+
+      assertRepr(getUpperBoundTag[Int].tag, "Set[=?: <Nothing..Int>]")
+      assertSame(getUpperBoundTag[Int].tag, Tag[Set[_ <: Int]].tag)
+
+      assertRepr(getLowerBoundTag[Int].tag, "Set[=?: <Int..Any>]")
+      assertSame(getLowerBoundTag[Int].tag, Tag[Set[_ >: Int]].tag)
+
+      // for recursive bound case
+      def x[T[_]: TagK  , U: Tag] = Tag[Set[_ <: T[U]]]
+      assertRepr(x[List, Long].tag, "Set[=?: <Nothing..List[+Long]>]")
+      assertSame(x[List, Long].tag, Tag[Set[_ <: List[Long]]].tag)
```

Comment:
@cmofx Given the following additional tests:

```suggestion
      assertSameStrict(x[List, Long].tag, Tag[Set[_ <: List[Long]]].tag)

      def xcov[T[_]: TagK, U: Tag] = Tag[List[_ <: T[U]]]
      assertRepr(xcov[List, Long].tag, "List[+?: <Nothing..List[+Long]>]")
      assertSameStrict(xcov[List, Long].tag, Tag[List[_ <: List[Long]]].tag)

      def xcontr[T[_]: TagK, U: Tag] = Tag[(_ <: T[U]) => Int]
      assertRepr(xcontr[List, Long].tag, "Function1[-?: <Nothing..List[+Long]>,+Int]")
      assertSameStrict(xcontr[List, Long].tag, Tag[(_ <: List[Long]) => Int].tag)
```

`xcov` case fails with
```
"List[+[List[+Long]]]" did not equal "List[+[?: <Nothing..List[+Long]>]]" (TagAssertions.scala:27)
```
Only on Scala versions 2.13 and 2.12 and works on Scala 2.11 and 3.3

Could you try to investigate whether this can be fixed?

Reply:
The xcov failed because the type bounds information is lost in
src/main/scala-2/izumi/reflect/TagMacro.scala, line 106, 
    val tgt = ReflectionUtil.norm(c.universe: c.universe.type, logger)(tpe.dealias)
where tpe.dealias removed the type bounds 

As dealias is implemented internally, I think it is not fixable.

Reply:
Yes, that does not seem to be reasonably fixable